PowerShellEditorServices.build.ps1

```diff
@@ -313,30 +313,50 @@ task RestorePsesModules -After Build {
     # Save each module in the modules.json file
     foreach ($moduleName in $moduleInfos.Keys)
     {
-        if (Test-Path -Path (Join-Path -Path $submodulePath -ChildPath $moduleName))
-        {
-            Write-Host "`tModule '${moduleName}' already detected. Skipping"
-            continue
+        $moduleInstallPath = (Join-Path -Path $subModulePath -ChildPath $moduleName)
+
+        try {
+            $moduleFolderChildItems = Get-ChildItem -Path $moduleInstallPath -Force
```

Comment:
I don't think this line does what you or at least what I think you think  :-)  `-Force` doesn't cause the non-terminating error that happens if `$moduleInstallPath` doesn't exist.  So the `catch` won't be invoked.  You could use `-ErrorAction Stop` to do that but in general I don't think it is a good idea to put anything but error-handling/logging in a `catch` statement.  I'd recommend leaving this logic in place and picking things up down at line 341.

Reply:
I use $moduleFolderChildItems later on and check for Count -eq 0, I included -Force to include hidden/system files.

Get-ChildItem throws an error if the path you're supplying doesn't exist, but I forgot it's non-terminating 😳 The reason I went this route is not to duplicate the logic for moving a module out of its versioned subfolder (once for existing modules and once for newly saved modules). But I also agree it's not best practice to put anything but actual error handling in a `catch` statement, so I'll change it.